pkg/provisioner/ironic/prepare_test.go

```diff
@@ -114,6 +114,21 @@ func TestPrepare(t *testing.T) {
 			defer inspector.Stop()
 
 			host := makeHost()
+			if i == 1 {
+				host.Spec.BMC.Address = "irmc://test.bmc/"
+				host.Status.Provisioning.RAID = &metal3v1alpha1.RAIDConfig{
+					HardwareRAIDVolumes: []metal3v1alpha1.HardwareRAIDVolume{
+						{
+							Name:  "root",
+							Level: "1",
+						},
+						{
+							Name:  "v1",
+							Level: "1",
+						},
+					},
+				}
+			}
```

Comment:
That's really a fragile approach, and moreover hides the setup in the cases list. In other situation we've used the builder pattern to allow each case to specify its own settings (as for the `testserver`), here's not yet available for the host, but at least an enrichment function can be added in the test struct, to be invoked on the host only when defined

Reply:
Thanks, I added a flag `existRaidConfig` to test cases.

Reply:
A good step in the right direction (at least more robust than before), even though the single test case readability is still not optimal (a builder approach would be definitely better); not anymore a blocking point for me though